src/main/scripts/src/app/shared/permissions/table/permissions-table.component.ts

```diff
@@ -44,57 +48,182 @@ export class PermissionsTableComponent implements OnInit {
   public entityType: string;
 
   public roles: string[];
-  public entities: Entity[];
+  public entities: Permission[];
   public otherEntities: string[];
 
-  public constructor(private rolesService: RolesService) {
+  public addedRoles: string[] = [];
+  public rolesCheckbox;
+
+  public constructor(private organizationService: OrganizationService,
+    private projectService: ProjectService) {
   }
 
   public ngOnInit() {
+    this.setPermissionsForResource(this.resourceType);
+
     this.roles = ROLES[this.resourceType];
-    this.entities = this.getEntities(this.entityType);
-    this.otherEntities = this.getOtherEntities(this.entityType);
+
+    this.rolesCheckbox = {};
+    this.roles.forEach(r => {
+      this.rolesCheckbox[r] = false;
+    });
   }
 
-  private getEntities(entityType: string): Entity[] {
-    switch (entityType) {
-      case 'users':
-        // TODO get from rolesService instead
-        return [
-          {name: 'marvec', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'livthomas', roles: ['read', 'share', 'write']}
-        ];
-      case 'groups':
-        // TODO get from rolesService instead
-        return [
-          {name: 'managers', roles: ['read', 'share', 'write', 'manage']},
-          {name: 'employees', roles: ['read', 'share']},
-          {name: 'contractors', roles: ['read']}
-        ];
-      default:
-        throw Error('unknown entity type');
+  private getEntities(entityType: string, permissions: Permissions): Permission[] {
+    let entities = permissions[entityType];
+    if (entities) {
+      return entities;
+    } else {
+      throw Error('unknown entity type');
+    }
+  }
+
+  private setPermissionsForResource(resourceType: string) {
+    let parent = this;
+    let getPermissionsFunctions = {
+      'organization': function() {return parent.organizationService.getPermissions();} ,
+      'project': function() {return parent.projectService.getPermissions();},
+      'collection': null, // TODO
+      'view': null // TODO
+    };
+    let getPermissions = getPermissionsFunctions[resourceType];
+    if (getPermissions) {
+      getPermissions().subscribe((permissions: Permissions) => {
+          this.entities = this.getEntities(this.entityType, permissions);
+          this.otherEntities = this.getOtherEntities(this.entityType);
+        });
+    } else {
+      throw Error('unknown resorce type');
     }
   }
 
   private getOtherEntities(entityType: string): string[] {
     switch (entityType) {
       case 'users':
         // TODO get from usersService instead
-        return ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
+        let userEntities = ['alicak', 'kubedo', 'jkotrady', 'kulexpipiens'];
```

Comment:
Can you extract user and group entities to some mockup of `userService`.. it doesn't have to work for now, just return the array. So that when implementing the service you just have to add proper implementation to the service instead of `return Observable.of(['alicak', 'kubedo', 'jkotrady', 'kulexpipiens']);` and not look for TODOs across the project.